py-polars/polars/internals/lazy_functions.py

```diff
@@ -1128,16 +1128,16 @@ def lit(
     """
     tu: TimeUnit
     if isinstance(value, datetime):
-        tu = "us"
-        e = lit(_datetime_to_pl_timestamp(value, tu)).cast(Datetime(tu))
+        tu = "us" if dtype is None else getattr(dtype, "tu", "us")
+        e = lit(_datetime_to_pl_timestamp(value, tu)).cast(dtype or Datetime(tu))
```

Comment:
is this correct
```python
In [8]: pl.select(pl.lit(datetime(2020, 1, 1, 0, 0), dtype=pl.Datetime('us', 'Asia/Kathmandu')))
Out[8]: 
shape: (1, 1)
┌──────────────────────────────┐
│ literal                      │
│ ---                          │
│ datetime[μs, Asia/Kathmandu] │
╞══════════════════════════════╡
│ 2020-01-01 05:45:00 +0545    │
└──────────────────────────────┘
```
?

I think I'd expect it to cast to the given timezone, rather than implicitly convert from UTC

Reply:
I've opened a PR to your PR with what I'd started: https://github.com/alexander-beedie/polars/pull/1

Reply:
Hmm... it's consistent with what we're doing everywhere else, as `_datetime_to_pl_timestamp` will forcibly convert to UTC. However, a very sensible follow-up to this might be to consider if that's really what we want to be doing, as that function looks like it doesn't convert to UTC, but uses `replace` - if a timezone was already set, I think we just trampled over it :\

Reply:
Sure thing

In meantime, perhaps do you want just change this to `.cast(Datetime(tu))`? I think it might be less surprising to keep time zones in `dtype` unsupported, rather than adding support here and then changing the behaviour later

Reply:
> Sure thing
> 
> In meantime, perhaps do you want just change this to `.cast(Datetime(tu))`? I think it might be less surprising to keep time zones in `dtype` unsupported, rather than adding support here and then changing the behaviour later

Can do; perhaps `_datetime_to_pl_timestamp` needs a tz param, so we can consolidate that behaviour...🤔

Reply:
I'll take a look 👀